test/files/neg/warn-unused-params.scala

```diff
@@ -151,3 +151,9 @@ object END
 class Nested {
   @annotation.unused private def actuallyNotUsed(fresh: Int, stale: Int) = fresh
 }
+
+class Annie(value: String) extends annotation.StaticAnnotation // no warn for annotation
+
+class Selfie {
+  def f(i: Int, j: Int) = this // no warn this is trivial
```

Comment:
I think this had a motivating example, but it looks wrong. `this` signals that the method is "merely" side-effecting, but why does that make it ok to ignore inputs?

Reply:
> Maybe it means "this implementation does trivially nothing, but an override may attempt to do real work.

I agree. What's the goal of the unused param warning?
1. when a param is unused, maybe the user referenced some other symbol by mistake
2. when refactoring and a param becomes unused, maybe it can be removed

In both cases, a trivial body like `this` is unlikely to be an accident